willa/tind/fetch.py

```diff
@@ -22,7 +24,35 @@ def fetch_metadata(record: str) -> Record:
         raise RecordNotFoundError(f"Record {record} not found in TIND.")
 
     records = parse_xml_to_array(StringIO(response))
-    if len(records) > 1:
-        raise RecordNotFoundError(f"Record {record} matched more than one record in TIND.")
+    if len(records) != 1:
+        raise RecordNotFoundError(f"Record {record} did not match exactly one record in TIND.")
 
     return records[0]
+
+
+def fetch_file(file_url: str, output_dir: str = ''):
+    """Fetch the given file from TIND.
+
+    :param str file_url: The URL to the file to download from TIND.
+                         This must be a TIND file download URL.
+    :param str output_dir: The directory in which to save the file.
+    :raises AuthorizationError: When the TIND API key is invalid, or the file is restricted.
+    :raises ValueError: When the URL is not a valid TIND file download URL.
+    :raises RecordNotFoundError: When the file is invalid or not found.
+    :raises IOError: When the file cannot be saved to the given output directory.
+    :returns: When the file has been successfully downloaded to the output directory.
+    """
+    if not file_url.endswith('/download/'):
+        raise ValueError('URL is not a valid TIND file download URL.')
+
+    if output_dir == '':
+        # This cannot be put as the default value for the parameter because it would set
+        # the value at the time this file is imported.  Changes in the environment (whether
+        # via config reload, the test infrastructure, etc) would not appear.
+        output_dir = os.getenv('DEFAULT_STORAGE_DIR')
+
+    output_filename = file_url.split('/')[-3]
```

Comment:
can we get this from the response headers instead of the URL?

Reply:
Updated with this behaviour.